shell.nix

```diff
@@ -35,8 +48,62 @@ in pkgs.mkShell {
   LANGUAGE = "en_US.UTF-8";
 
   QTDIR = qtCustom;
+# TODO: still needed?
+# https://github.com/NixOS/nixpkgs/pull/109649
+  QT_INSTALL_PLUGINS = "${qtCustom}/${pkgs.qt515.qtbase.qtPluginPrefix}";
 
   shellHook = ''
-    export MAKEFLAGS="-j$NIX_BUILD_CORES"
-  '';
+    export PATH="${pkgs.lddWrapped}/bin:$PATH"
+    '';
+
+# Used to workaround missing lib links in qt-custom
+# TODO:check again after Qt upgrade
+  LIBRARY_PATH = with pkgs.qt515; pkgs.lib.makeLibraryPath [
+    qtdeclarative
+    qtquickcontrols
+    qtquickcontrols2
+    qtsvg
+    qtmultimedia
+    qtwebview
+    qtwebchannel
+  ];
+
+# Used for linuxdeployqt
+# TODO:check again after Qt upgrade
+  LD_LIBRARY_PATH = with pkgs; lib.makeLibraryPath [
+    qt515.qtquickcontrols2
+    qt515.qtdeclarative
+    qt515.qtmultimedia
+    qt515.qtbase
+    qt515.qtsvg
+    qt515.qtwebengine
+    qt515.qtlocation
```

Comment:
There's no need for every package to have its own line. Also you can use `with qt515;` to avoid repeating yourself.

Reply:
Actually why not just add contents of `LIBRARY_PATH` here? Or even make a separate list since I can see this being repeated in a few places.

Reply:
I've cleaned up `LIBRARY_PATH` fix - it is not needed anymore with new Qt derivation.